ast/env.go

```diff
@@ -253,6 +270,12 @@ func (n *typeTreeNode) Put(path Ref, tpe types.Type) {
 	curr.value = tpe
 }
 
+func (n *typeTreeNode) PutArg(path Ref, i int, tpe types.Type) {
+	pos := json.Number(strconv.Itoa(i))
```

Comment:
Storing the arguments in the TypeEnv this way is clever but I'm not sure it's a good idea to use refs this way. Before, refs referred to documents. With this change, that's no longer entirely true. How would you refer to the the output of a function? It seems like this would be ambiguous w/ overloaded functions.